src/Rules/Comparison/BooleanXorConstantConditionRule.php

```diff
@@ -0,0 +1,59 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Comparison;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\BinaryOp\LogicalXor;
+use PHPStan\Analyser\Scope;
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleErrorBuilder;
+use PHPStan\Type\Constant\ConstantBooleanType;
+use function sprintf;
+
+/**
+ * @implements Rule<LogicalXor>
+ */
+class BooleanXorConstantConditionRule implements Rule
+{
+
+	public function __construct(
+		private bool $treatPhpDocTypesAsCertain,
+	)
+	{
+	}
+
+	public function getNodeType(): string
+	{
+		return LogicalXor::class;
+	}
+
+	public function processNode(Node $node, Scope $scope): array
+	{
+		$messages = [];
+		$tipText = 'Because the type is coming from a PHPDoc, you can turn off this check by setting <fg=cyan>treatPhpDocTypesAsCertain: false</> in your <fg=cyan>%configurationFile%</>.';
```

Comment:
Sweet, i wasn't sure of the role of the `ConstantConditionRuleHelper`, as it looked like it was primarily used in checking for "always x" values on the left or right, i thought it might not be necessary here.

I'll look into using it, although i'm not 100% sure on how to apply it so may need a bit of guidance later!

Reply:
Exactly 😊